internal/ingest/ingest.go

```diff
@@ -869,15 +874,19 @@ func (ing *Ingester) ingestWorkerLogic(ctx context.Context, provider peer.ID) {
 		ctxIdStr := string(ai.ad.ContextID)
 		// This ad was deleted by a later remove. Push previous onto skips
 		// stack, and set latest skip.
-		if _, ok := rmCtxID[ctxIdStr]; ok {
+		if rmCtxID[ctxIdStr] || ing.deletedProviders[ai.ad.Provider] {
 			skips = append(skips, skip)
 			skip = i
 			continue
 		}
 		// If this is a remove, do not skip this ad, but skip all earlier
 		// (deeped in chain) ads with the same context ID.
 		if ai.ad.IsRm {
-			rmCtxID[ctxIdStr] = struct{}{}
+			if len(ai.ad.ContextID) != 0 {
+				rmCtxID[ctxIdStr] = true
+			} else {
+				ing.deletedProviders[ai.ad.Provider] = true
```

Comment:
When an ad to delete a provider is seen, handle that ad immediately, because it is the only one in the chain we new to handle for this provider. None of this provider's other ads matter since the provider is getting deleted.
```suggestion
				// Mark the provider as deleted to skip any other ads.
				deletedProviders[ai.ad.Provider] = true
				// Process this ad now, since it is the only one that
				// matters for the provider that is being deleted.
				if markErr := ing.markAdProcessed(assignment.publisher, ai.cid); markErr != nil {
					log.Errorw("Failed to mark ad as processed", "err", markErr)
				}
				ing.inEvents <- adProcessedEvent{
					publisher: assignment.publisher,
					headAdCid: assignment.adInfos[0].cid,
					adCid:     ai.cid,
				}
			}
```